packages/site/src/components/PropList.js

```diff
@@ -0,0 +1,205 @@
+    textAlign: 'left'
+  }),
+  propP: {
+    margin: 0
+  },
+  propName: (props, theme) => ({
+    fontFamily: theme.fontFamily_monospace
+  }),
+  propRequired: (props, theme) => ({
+    color: theme.color_background_danger,
+    fontFamily: theme.fontFamily_monospace
+  }),
+  propRow: (props, theme) => ({
+    borderBottom: `1px solid ${theme.color_gray}`
+  }),
+  propTable: {
+    borderCollapse: 'collapse',
+    borserSpacing: 0,
+    width: '100%'
+  },
+  propType: (props, theme) => ({
+    display: 'inline-block',
+    backgroundColor: theme.color_gray,
+    borderRadius: '3px',
+    fontFamily: theme.fontFamily_monospace,
+    padding: theme.measurement_a
+  }),
+  propValue: (props, theme) => ({
+    border: `1px solid ${theme.color_gray}`,
+    height: '100%',
+    resize: 'vertical',
+    width: '100%'
+  }),
+  root: (props, theme) => ({
+    fontFamily: theme.fontFamily_system,
+    margin: `${theme.measurement_d} 0 0 0`
+  })
+};
+
+const CodeValue = createStyledComponent('span', styles.codeValue);
+const PropTable = createStyledComponent('table', styles.propTable);
+const PropColumnHeader = createStyledComponent('th', styles.propColumnHeader);
+const PropCell = createStyledComponent('td', styles.propCell);
+const PropName = createStyledComponent('span', styles.propName);
+const PropP = createStyledComponent('p', styles.propP);
+const PropRequired = createStyledComponent('span', styles.propRequired);
+const PropRow = createStyledComponent('tr', styles.propRow);
+const PropType = createStyledComponent('span', styles.propType);
+const PropValue = createStyledComponent('textarea', styles.propValue);
+const Root = createStyledComponent('div', styles.root);
+
+function DefaultValue({
+  defaultValue,
+  required
+}: {
+  defaultValue?: any,
+  required?: boolean
+}) {
+  return required
+    ? <PropRequired>required</PropRequired>
+    : <CodeValue>
+        {undefined === defaultValue
+          ? 'undefined'
+          : JSON.stringify(defaultValue)}
+      </CodeValue>;
+}
+
+type OnPropChange = (propDescription: Object, newValue: any) => void;
+
+type DocProps = {
+  defaultValue?: any,
+  description?: string,
+  exampleValue?: any,
+  name: string,
+  onPropChange: OnPropChange,
+  type: string,
+  required?: boolean
+};
+
+function onPropValueChange(
+  e: Object,
+  propDescription: Object,
+  callback: OnPropChange
+) {
+  try {
+    const newValue = JSON.parse(e.target.value);
+    callback(propDescription, newValue);
+  } catch (e) {
+    console.warn('invalid JSON string');
+  }
+}
+
+function PropDoc({
+  defaultValue,
+  description,
+  exampleValue,
+  name,
+  onPropChange,
+  required,
+  type
+}: DocProps) {
+  return (
+    <PropRow>
+      <PropCell><PropName>{name}</PropName></PropCell>
+      <PropCell>
+        <DefaultValue defaultValue={defaultValue} required={required} />
+      </PropCell>
+      <PropCell>
+        <PropP><PropType>{type}</PropType>&nbsp;{description}</PropP>
```

Comment:
I believe that `type` should be a separate field from `description` and we should always attempt to document all props.  Separate columns improves the readability for me and it seems to be the most common pattern across various component libraries.